src/softmax.cpp

```diff
@@ -33,8 +33,55 @@
 #include <miopen/softmax/solvers.hpp>
 #include <miopen/find_solution.hpp>
 
+#include <nlohmann/json.hpp>
+
 namespace miopen {
 
+extern "C" miopenStatus_t miopenCreateSoftmaxDescriptor(miopenSoftmaxDescriptor_t* activDesc)
+{
+
+    MIOPEN_LOG_FUNCTION(activDesc);
+    return miopen::try_([&] { miopen::deref(activDesc) = new miopen::SoftmaxDescriptor(); });
```

Comment:
```suggestion
    return miopen::try_([&] {
        auto& desc = miopen::deref(activDesc);
        desc = new miopen::SoftmaxDescriptor();
    });
```
This way memory would not leak if deref throws